streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -905,18 +913,24 @@ private long computeLatency() {
 
     /**
      * Shutdown this stream thread.
+     * Note that there is nothing to prevent this function from being called multiple times
+     * (e.g., in testing), hence the state is set only the first time
      */
     public synchronized void close() {
         log.info("{} Informed thread to shut down", logPrefix);
         setState(State.PENDING_SHUTDOWN);
     }
 
     public synchronized boolean isInitialized() {
-        return state == State.RUNNING;
+        synchronized (stateLock) {
```

Comment:
we can remove `synchronized` from line 924. Also i'm wondering if we really need to synchronize on the `stateLock` here? The `state` variable is volatile, so other threads will immediately see the changes. Also i don't think it buys anything as even if you get the `stateLock` and it returns `true` the Thread could immediately transition to a non running state.